### ISSUE TITLE
Add option for directly viewing pdf files

### ISSUE THREAD

**Issue by sanjitchak** [amahi/android#352]
Right now, Amahi Android app can open pdf files using external apps. 

But since, pdf viewer can be integrated in-app,
we can add a feature of directly opening **pdf** files in the app only.

Implementation:-
Using [AndroidPdfViewer](https://github.com/barteksc/AndroidPdfViewer)

**Comment by 207ashu** [comment]
I'm new to Open Source, Can I contribute to this project? I have worked on AndroidPdfViewer earlier also.

**Comment by cpg** [comment]
Definitely, anyone can, it's Open Source!

The only concern is how much does something like this add to the binary. Any guesses?

If it's a substantial amount, it would be hard to justify, since I would imagine most people have PDF readers already in their phones.

**Comment by 207ashu** [comment]
This increases the size of the apk by approximately 12 Mbs

**Comment by cpg** [comment]
Thanks for the estimate. That's about 15% increase in size, which is kind of noticeable. It makes is large that some Google apps.

I wonder about the features. I don't know enough about people using Android apps for PDF.

Any other opinions on this?

**Comment by sanjitchak** [comment]
I raised this issue so that very *sensitive* pdf can be opened in the app only as old android phones are prone to hacks.
So, if most of our users, use the android app for fun and videos then I think that increase in size may not be justified. Most android phones come with pdf viewer preinstalled.

**Comment by codedentwickler** [comment]
I think it's a good thing to have a way to open pdf in the app like @sanjitchak said for sensitive files. I don't see a need to add a library with a 12 MBs overhead to achieve this. What if a Web view is used to preview the document in the app, fancy features won't be available though. We can then allow the user to open with their favorite app from the Webview (maybe through the menu option). This implementation won't result to a tangible apk size increase.

**Comment by cpg** [comment]
Good idea. The webview solution may work as a good compromise. I think any recent version of the webview probably can handle PDFs, it's built in, and if we can make it that the user can also open it with other apps, that would work well.

**Comment by sanjitchak** [comment]
Yes, for opening with other apps, user may have the option of downloading the pdf. Maybe, a warning message if the user clicks on download.

**Comment by chirag-jn** [comment]
Are we still planning to open the PDFs in an external viewer or can I start working on opening the PDF in a webview?

**Comment by shivamparashar165** [comment]
Love to work on this, I can do it.

Give me a chance
